modules/calc/src/main/java/com/opengamma/strata/calc/runner/function/result/CurrencyValuesArray.java

```diff
@@ -94,6 +96,32 @@ public static CurrencyValuesArray of(List<CurrencyAmount> amounts) {
     return new CurrencyValuesArray(currency, DoubleArray.ofUnsafe(values));
   }
 
+  /**
+   * Obtains an instance using a function to create the entries.
+   * <p>
+   * The function is passed the scenario index and returns the value for that index.
+   * 
+   * @param size  the number of elements
+   * @param valueFunction  the function used to obtain each value
+   * @return an instance initialized using the function
+   * @throws IllegalArgumentException is size is zero or less
+   */
+  public static CurrencyValuesArray of(int size, IntFunction<CurrencyAmount> valueFunction) {
+    ArgChecker.notNegativeOrZero(size, "size");
```

Comment:
Since the currency is extracted from the function, the size must be at least size one.